tests/testthat/test-github_workflow_present.R

```diff
@@ -0,0 +1,30 @@
+make_github_workflow <- function(source, source_workflow_path) {
+  template_path <- "../../inst/github_workflow_template.yaml"
+  template <- readLines(template_path)
+  newfile <- gsub("_SOURCE_", source, template)
+  writeLines(newfile, source_workflow_path)
+  message(
+    paste("workflow auto created for", source, "at", source_workflow_path)
+  )
+}
+
+test_github_workflow_found <- function(source) {
+  source_workflow_path <- paste0("../../.github/workflows/", source, ".yaml")
+  if (!file.exists(source_workflow_path)) {
```

Comment:
part of me thinks this is a bit strange as the test can never fail (as the file will be created). Makes me think it should be a manual function package devs have to call... this might also be nice for people wanting to set up their own testing frameworks (in line with the test PR) but doesn't solve the what if I forget problem (but having a test to check if it is there does that).

Reply:
Yeah I thought about having the test fail but showing a message telling them to call something like `make_github_workflow()` but then if its that easy why not just do it automatically? I agree though it is a bit weird it being in the tests but I wasn't sure where to put something that would do this automatically. Happy to move this to someplace else where it will get called when adding a class

Reply:
I think my preference is to have it as a function that takes a target directory for the action (defaulting to `.github/workflows` then make the user manually call it + check they have with tests. Slightly more annoying but always someone to use `DataClass` elsewhere with minimal effort which I think is perhaps nice. In the future we could have a function that writes a `DataClass` scaffold for someone to use for a new dataset + makes this action. That would be really neat as we think about using the tooling elsewhere perhaps.